test/api_integration/apis/search/search.ts

```diff
@@ -17,72 +17,101 @@
  * under the License.
  */
 
+import expect from '@kbn/expect';
 import { FtrProviderContext } from '../../ftr_provider_context';
 
 export default function ({ getService }: FtrProviderContext) {
   const supertest = getService('supertest');
 
-  describe('msearch', () => {
+  describe('search', () => {
     describe('post', () => {
-      it('should return 200 when correctly formatted searches are provided', async () =>
-        await supertest
-          .post(`/internal/_msearch`)
+      it('should return 200 when correctly formatted searches are provided', async () => {
+        const resp = await supertest
+          .post(`/internal/search/es`)
           .send({
-            searches: [
-              {
-                header: { index: 'foo' },
-                body: {
-                  query: {
-                    match_all: {},
-                  },
+            params: {
+              body: {
+                query: {
+                  match_all: {},
                 },
               },
-            ],
+            },
           })
-          .expect(200));
+          .expect(200);
 
-      it('should return 400 if you provide malformed content', async () =>
-        await supertest
-          .post(`/internal/_msearch`)
-          .send({
-            foo: false,
-          })
-          .expect(400));
+        expect(resp.body.isPartial).to.be(false);
+        expect(resp.body.isRunning).to.be(false);
+        expect(resp.body).to.have.property('rawResponse');
+      });
 
-      it('should require you to provide an index for each request', async () =>
+      it('should return 404 when if no strategy is provided', async () =>
         await supertest
-          .post(`/internal/_msearch`)
+          .post(`/internal/search`)
           .send({
-            searches: [
-              { header: { index: 'foo' }, body: {} },
-              { header: {}, body: {} },
-            ],
+            body: {
+              query: {
+                match_all: {},
+              },
+            },
           })
-          .expect(400));
+          .expect(404));
 
-      it('should not require optional params', async () =>
-        await supertest
-          .post(`/internal/_msearch`)
+      it('should return 500 when if no strategy is provided', async () => {
+        const resp = await supertest
+          .post(`/internal/search/banana`)
           .send({
-            searches: [{ header: { index: 'foo' }, body: {} }],
+            body: {
+              query: {
+                match_all: {},
+              },
+            },
           })
-          .expect(200));
+          .expect(500);
```

Comment:
Hmm, there are some interesting cases here... We should probably do a 400 when an invalid search strategy is given or an `indexType` is provided in OSS since those are really bad requests. Thoughts?

Reply:
Fixed!